### ISSUE TITLE
Test diffrent compilers in CI

### ISSUE THREAD

**Issue by Mikolaj-A-Kowalski** [CambridgeNuclear/SCONE#42]
Currently the `build-and-test` workflow only compiles SCONE with the default Fortran compiler of Ubuntu 20.04LTS (gfortran-11). However we would probably wish to test with more versions to see if there are any problems. In particular: 
- gfortran-8 (which will become the lowest supported version soon
- gfortran-12 (or whichever is the latest version at the moment) 

For good measure we could just cover them all for `>8`. 

Basically I think we should aim to have a similar solution we had on BitBucket, with containers having the right python for pfUnit, compiler and pfUnit preinstalled (to save build time). 

I will try to set it up soon.

**Comment by Mikolaj-A-Kowalski** [comment]
Resolved by #45
